extensions/parquet/table/src/test/java/io/deephaven/parquet/table/ParquetTableReadWriteTest.java

```diff
@@ -665,14 +698,84 @@ public void groupingColumnsBasicWriteTestsImpl(TestParquetTableWriter writer) {
         }
 
         // Make sure that original file is preserved and no temporary files
-        filesInDir = Arrays.asList(parentDir.list());
-        assertTrue(filesInDir.size() == 2 && filesInDir.contains(destFilename)
-                && filesInDir.contains(vvvGroupingFilename));
+        verifyFilesInDir(parentDir, new String[] {destFilename}, Map.of("vvv", new String[] {vvvGroupingFilePath}));
         fromDisk = ParquetTools.readTable(destFile);
         TstUtils.assertTableEquals(fromDisk, tableToSave);
         FileUtils.deleteRecursively(parentDir);
     }
 
+    @Test
+    public void parquetDirectoryWithHiddenFilesTest() throws IOException {
+        // Create an empty parent directory
+        final File parentDir = new File(rootFile, "tempDir");
+        parentDir.mkdir();
+        assertTrue(parentDir.exists() && parentDir.isDirectory() && parentDir.list().length == 0);
+
+        Integer data[] = new Integer[500 * 4];
+        for (int i = 0; i < data.length; i++) {
+            data[i] = i / 4;
+        }
+        final TableDefinition tableDefinition = TableDefinition.of(ColumnDefinition.ofInt("vvv").withGrouping());
+        final Table tableToSave = TableTools.newTable(tableDefinition, TableTools.col("vvv", data));
+
+        final String destFilename = "data.parquet";
+        final File destFile = new File(parentDir, destFilename);
+        ParquetTools.writeTable(tableToSave, destFile);
+        String vvvGroupingFilePath = ".dh_metadata/indexes/vvv/index_vvv_data.parquet";
+        // Hidden metadata in parent directory
+        verifyFilesInDir(parentDir, new String[] {destFilename}, Map.of("vvv", new String[] {vvvGroupingFilePath}));
+
+        // Call readTable on parent directory
+        Table fromDisk = ParquetTools.readTable(parentDir);
+        TstUtils.assertTableEquals(fromDisk, tableToSave);
+
+        // Add an empty dot file in the parent directory
+        final File dotFile = new File(parentDir, ".hiddenData");
+        assertTrue(dotFile.createNewFile());
+        fromDisk = ParquetTools.readTable(parentDir);
+        TstUtils.assertTableEquals(fromDisk, tableToSave);
+
+        // Add another table in a hidden file in parent directory
+        final Table anotherTable = TableTools.emptyTable(5).update("A=(int)i");
+        final File pqDotFile = new File(parentDir, ".hiddenData.parquet");
```

Comment:
This will fail on OS where `.` prefixes aren't honored as hidden (i.e. Windows is stupid).